README.md

```diff
@@ -16,340 +33,655 @@ $ pip install py-sc-client
+
 Search by sc-template system identifier.
-```py
+
+```python
+from sc_client.client import template_search
+from sc_client.models import ScAddr
+
+link_node: ScAddr
 search_params = {'_link': link_node, '_var_node': 'node_idtf'}
-search_results = client.template_search('my_template', search_params)
+search_results = template_search('my_template', search_params)
 search_result = search_results[0]
 
-len(search_result)
 ```
 
 Search by scs sc-template.
-```py
-search_results = client.template_search('class _-> _node;;')
-search_result = search_results[0]
 
-len(search_result)
+```python
+from sc_client.client import template_search
+
+search_results = template_search('class _-> _node;;')
+search_result = search_results[0]
 ```
 
-### template_generate
-Generate a construction in the KB memory by template.
+#### Generate template
 
-*Parameters*: An ScTemplate class object or a scs-template as a string, an ScTemplateParams class object.
-*Returns*: An ScTemplateResult class object.
+- *sc_client.client*.**template_generate**(template: ScTemplate, params: ScTemplateParams = None)
 
-```py
-templ = ScTemplate()
-templ.triple_with_relation( # faaaf
-    (main_node, '_main_node'),
+Returns ScTemplateResult by *template*.
+
+```python
+from sc_client.client import template_generate
+from sc_client.constants import sc_types
+from sc_client.models import ScTemplate, ScAddr
+
+main_node: ScAddr
+relation_node: ScAddr
+link_node: ScAddr
+# Some ScAddrs for example
+
+template = ScTemplate()
+template.triple_with_relation(
+    main_node >> '_main_node',
     sc_types.EDGE_D_COMMON_VAR,
-    [sc_types.LINK_VAR, '_link'],  # Do not use List aliased item, use tuple instead (deprecated 0.3.0)
+    sc_types.LINK_VAR >> '_link',
     sc_types.EDGE_ACCESS_VAR_POS_PERM,
     relation_node,
 )
-templ.triple( # faa
+template.triple(
     '_main_node',
     sc_types.EDGE_ACCESS_VAR_POS_TEMP,
     (sc_types.NODE_VAR, '_var_node')
 )
 gen_params = {'_link': link_node, '_var_node': 'node_idtf'}
-gen_result = client.template_generate(templ, gen_params)
+gen_result = template_generate(template, gen_params)
 ```
 
-Also, you can generate a construction by template address or its system identifier or scs-template as well as search constructions.
+Also, you can generate a construction by template address or its system identifier or scs-template as well as search
+constructions.
 
-### events_create
-Create an event in the KB memory.
+#### ScTemplateResult
+
+After operations with template you'll receive ScTemplateResult:
+
+- *sc_client.models*.**ScTemplateResult**
+
+It has all ScAddr and aliases. You can get addrs or iterate.
+
+Methods:
+
+- **len**(*ScTemplateResult*)
+
+  Get count of elements
+- *ScTemplateResult*.**size**()
+
+  The same, deprecated in version 0.3.0
+- *ScTemplateResult*[i: int]
+
+  Get ScAddr by index
+- *ScTemplateResult*.**get**(i: int)
+
+  The same, deprecated in version 0.3.0
+- *ScTemplateResult*.**get**(alias: str)
```

Comment:
Don't deprecate get by index, it is needed when i want to get all addrs of elements in construction

Reply:
Removed deprecated mark